knative-operator/pkg/controller/knativeserving/kourier/kourier.go

```diff
@@ -40,14 +33,32 @@ func apply(instance *servingv1alpha1.KnativeServing, api client.Client, path str
 
 // Delete deletes Kourier resources.
 func Delete(instance *servingv1alpha1.KnativeServing, api client.Client) error {
-	for _, path := range manifests {
-		manifest, err := mf.NewManifest(path, false, api)
-		if err != nil {
-			return err
-		}
-		if err := manifest.DeleteAll(); err != nil {
-			return err
-		}
+	log.Info("Deleting Kourier Ingress")
+	manifest, err := mf.NewManifest(path, false, api)
+	if err != nil {
+		return err
+	}
+	transforms := []mf.Transformer{mf.InjectNamespace(ingressNamespace(instance.GetNamespace()))}
+
+	if err := manifest.Transform(transforms...); err != nil {
+		return err
 	}
-	return nil
+	if err := manifest.DeleteAll(); err != nil {
+		return err
+	}
+
+	log.Info("Deleting ingress namespace")
+	ns := &v1.Namespace{}
+	err = api.Get(context.TODO(), client.ObjectKey{Name: ingressNamespace(instance.GetNamespace())}, ns)
```

Comment:
I expected the namespace is removed by `DeleteAll()`, but it didn't due to manifestival's logic. I think that it is expected but opened the report here https://github.com/manifestival/manifestival/issues/8 FYI @jcrossley3

Reply:
Commented on the issue, thanks, but wanted to mention that manifestival will only delete namespaces it originally creates. Or at least, that's how it's supposed to work. Because it has no idea what other resources might be in that ns, so doesn't want to surprise anyone by deleting them. Maybe that's flawed logic?

Reply:
Thank you for the detail. Then, we need this deletion of knative-serving-ingress ns code anyway.